source/common/mongo/proxy.cc

```diff
@@ -241,5 +245,54 @@ DecoderPtr ProdProxyFilter::createDecoder(DecoderCallbacks& callbacks) {
   return DecoderPtr{new DecoderImpl(callbacks)};
 }
 
+FaultConfig::FaultConfig(const Json::Object& fault_config)
+    : delay_percent_(
+          static_cast<uint32_t>(fault_config.getObject("fixed_delay")->getInteger("percent"))),
+      duration_ms_(
+          static_cast<uint64_t>(fault_config.getObject("fixed_delay")->getInteger("duration_ms"))) {
+}
+
+Optional<uint64_t> ProxyFilter::delayDuration() {
+  Optional<uint64_t> result;
+
+  if (!fault_config_)
+    return result;
+
+  if (!runtime_.snapshot().featureEnabled("mongo.fault.delay.percent",
+                                          fault_config_->delayPercent())) {
+    return result;
+  }
+
+  uint64_t duration = runtime_.snapshot().getInteger("mongo.fault.delay.duration_ms",
```

Comment:
similar to fault filter these keys should be static strings ex: https://github.com/lyft/envoy/blob/master/source/common/http/filter/fault_filter.cc#L24

Reply:
we are somewhat inconsistent with strings for runtime keys across the envoy code base, no objection to make that const.